### PR TITLE
Use canonical (browser-compatible) UserAgent format: spaces instead of commas

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

**Comment by ash211** [comment]
If UserAgent is now RFC-compatible can you link to that RFC in Javadoc?

File: http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java [hunk 1]
```diff
@@ -88,83 +87,76 @@ private static String formatSingleAgent(UserAgent.Agent agent, Map<String, Strin
...
+    private static Map<String, String> parseComments(String commentsString) {
+        Map<String, String> comments = Maps.newHashMap();
+        for (String comment : COMMENT_SPLITTER.split(commentsString)) {
+            List<String> fields = COLON_SPLITTER.splitToList(comment);
+            if (fields.isEmpty()) {
+                // continue
+            } else if (fields.size() == 2) {
+                comments.put(fields.get(0), fields.get(1));
+            } else {
+                comments.put(comment, comment);
```
Comment (ellisjoe) [inline http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java:157]: MapSplitter doesn't work because of this case?

**Comment by uschi2000** [comment]
linked

File: http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java [hunk 2]
```diff
@@ -176,7 +168,7 @@ static boolean isValidNodeId(String instanceId) {
     }
 
     static boolean isValidVersion(String version) {
```
Comment (ellisjoe) [inline http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java:171]: when is this used?

File: http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java [hunk 3]
```diff
@@ -88,83 +87,76 @@ private static String formatSingleAgent(UserAgent.Agent agent, Map<String, Strin
...
     private static UserAgent parseInternal(String userAgent, boolean lenient) {
         ImmutableUserAgent.Builder builder = ImmutableUserAgent.builder();
-        List<String> parts = COMMA_SPLITTER.splitToList(userAgent);
-        if (parts.isEmpty()) {
-            if (lenient) {
-                log.warn("Empty user agent, falling back to default/unknown agent");
-                return builder.primary(UserAgent.Agent.of("unknown", UserAgent.Agent.DEFAULT_VERSION)).build();
+
+        Pattern segmentPattern = Pattern.compile(
+                String.format("(%s)/(%s)( \\((.+?)\\))?", NAME_REGEX, LENIENT_VERSION_REGEX));
```
Comment (ellisjoe) [inline http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java:111]: fun

**Review by ellisjoe — APPROVED** [review]

File: http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java (hunk 1 above)
Comment (uschi2000) [reply http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java:157]: yep

File: http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java (hunk 2 above)
Comment (uschi2000) [reply http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java:171]: used in Agent builder

Comment (uschi2000) [reply http-clients/src/main/java/com/palantir/remoting3/clients/UserAgents.java:171]: to turn unknown versions into 0.0.0

**Pull request merged**